ethtx/providers/semantic_providers/database.py

```diff
@@ -40,13 +40,25 @@ def get_collection_count(self) -> int:
         return len(self._db.list_collection_names())
 
     @cache
-    def get_address_semantics(self, chain_id, address) -> Optional[Dict]:
+    def get_address_semantics(
+        self, chain_id, address, *, cursor_timeout_millis=None
+    ) -> Cursor:
```

Comment:
if this return cursor, this can't be cache. If it's working, it's only because we got lucky because `cache`  is getting all the data from this cursor. Could we add `list(...)` around returned value, and change type from `Cursor`, just to make it clear what is returned?